### PR TITLE
KimYi fixes

### PR DESCRIPTION
**Pull request by ltratt**
A couple of relatively small fixes and updates to the KimYi algorithm. The first commit is part of our extension of KimYi's algorithm; the second and third commits make us more faithfully follow the paper.

### CODE REVIEW

**Comment by ptersilie** [comment]
Ironically the commit talking about redundancy has a redundant word in it:
`Redundant configurations in the paper are determined solely determined by the
parsing stack`

File: src/lib/kimyi.rs [hunk 1]
```diff
@@ -655,8 +657,7 @@ E: 'OPEN_BRACKET' E 'CLOSE_BRACKET'
         assert_eq!(errs[0].repairs().len(), 1);
         check_repairs(&grm,
                       errs[0].repairs(),
-                      &vec!["InsertTerm \"A\", InsertTerm \"CLOSE_BRACKET\", InsertTerm \"CLOSE_BRACKET\"",
-                            "InsertTerm \"B\", InsertTerm \"CLOSE_BRACKET\", InsertTerm \"CLOSE_BRACKET\""]);
+                      &vec!["InsertNonterm \"E\", InsertTerm \"CLOSE_BRACKET\", InsertTerm \"CLOSE_BRACKET\""]);
```
Comment (ptersilie) [inline src/lib/kimyi.rs:660]: Why does renaming `A` to `E` needs removing `B`?

**Comment by ptersilie** [comment]
Two minor comments. Rest looks good.

File: src/lib/kimyi.rs (hunk 1 above)
Comment (ltratt) [reply src/lib/kimyi.rs:660]: That's not really what's going on here: it's the change from `InsertTerm` to `InsertTerm`. That's a knock-on effect of changing the hashing algorithm. Put another way, the hashing algorithm affects which repairs are considered equivalent to each other; changing it implies that different repairs may be found (as in the case here).

**Comment by ltratt** [comment]
OK, I've fixed the commit text, and (hopefully) explained why the other commit isn't a problem. If you're OK with this, I think it can be merged.

File: src/lib/kimyi.rs (hunk 1 above)
Comment (ptersilie) [reply src/lib/kimyi.rs:660]: Ah, missed the `InsertNonterm`.

**Comment by ptersilie** [comment]
Does this need any squashing?

**Pull request merged**